README
Epson Inkjet Printer Driver (ESC/P-R) Manual

Copyright (C) Seiko Epson Corporation 2005-2017.

Please visit our website for the latest information.
Please refer to follows for the information about this driver
http://download.ebz.epson.net/dsc/search/01/search/?OSC=LX


Contents
1  License Agreement
2  Revision History
3  Product Description
4  Installation and CUPS setting
    4.1  Installing
    4.2  Settings after installation
5  Printing
    5.1  Items to be set
    5.2  Details of setting items
    5.3  How to Set
    5.4  How to Print
6  How to build in non-LSB distribution
    6.1  For Redhat based distributions
    6.2  For Debian based distributions
    6.3  Other distributions
7  Acknowledgments


-------------------------------------------------------------------------------
1  License Agreement
-------------------------------------------------------------------------------
This software is covered by the GNU General Public License (GPL). 
Refer to the file "COPYING" for more information about the GPL.

-------------------------------------------------------------------------------
2  Revision History
-------------------------------------------------------------------------------
See ./NEWS.


-------------------------------------------------------------------------------
3  Product Description
-------------------------------------------------------------------------------
This software is a filter program used with the Common UNIX Printing
System (CUPS) on the Linux platform.
The software offers high quality printing with Seiko Epson color ink jet printers.

This product can only be used with printers that support the Epson ESC/P-R language.

Please see refer to follows for a list of supported printers.
http://download.ebz.epson.net/dsc/search/01/search/?OSC=LX

This software is based on LSB 3.2. It can be used in distribution supporting LSB 3.2 later.
If it is used in non-LSB distribution, you can use it by building from the source code.

-------------------------------------------------------------------------------
4  Installation and CUPS Setting
-------------------------------------------------------------------------------
4.1  Installation
	4.1.1  In the case of LSB supported distributions
	   Choose the appropriate command for your distribution from the list below
	   and execute it with root privileges.
	   Note: The following package name differs depending on the version
	   you are using.
	   For Redhat based distributions:
	   -----------------
	   # rpm -i epson-inkjet-printer-escpr-1.6.18-1lsb3.2.i486.rpm
	   -----------------
	   For Debian based distributions:
	   -----------------
	   # dpkg -i epson-inkjet-printer-escpr_1.6.18-1lsb3.2_i386.deb
	   -----------------
	   
	4.1.2  In the case of non-LSB supported distributions
	   Refer to "6  How to build in non-LSB distribution".


4.2  Settings after installation
   CUPS must be set to enable use of this software.
   Before printing, be sure to carry out the following steps.

	4.2.1  Restart CUPS
	     Restart CUPS, to enable the installed filter.
	     Execute the following command with root privileges.

	     Example: With Red Hat Linux
	     -----------------
	     # /etc/init.d/cups restart
	     -----------------
	     With Debian GNU/Linux
	     -----------------
	     # /etc/init.d/cupsys restart
	     -----------------

	4.2.2  Add a printer

	  Using commands
	     Execute the following command with root privileges.
	     -----------------
	     # lpadmin -p [PRINTER_NAME] -v [DEVICE_URI] -P [PPD_FILE] -E
	     -----------------

	     Options: -p      Specifies the setting name of the printer
	                        -E      Enables the printer
	                        -v      Specifies the driver and connection
	                        -P      Specifies the setting file (PPD file)

	     Example: With the PM-A950 printer 
	     -----------------
	     # lpadmin -p ep901a -v usb:/EPSON/EP-901A -P /opt/epson-inkjet-printer-escpr/ppds/EPSON/EPSON-EP-901A-epson-escpr-en.ppd.gz -E
	     -----------------

	     For details of lpadmin, see lpadmin(8).


	  Using a browser
	     CUPS is provided with a CGI for settings so that you can set your printer
	     using a browser. To add a printer, follow the steps below.
	       Note: The following menu items might differ depending on the CUPS
	             version and the environment used.

	     (1) Turn on the printer, and please be connected to the PC with a USB cable.
	     (2) Connect to CUPS Server (Port: 631) with the browser
	         Example: When CUPS is on localhost, access the following URL.
	             -----------------
	             http://localhost:631/
	             -----------------
	     (3) Select "Add Printer"
	     (4) Enter a printer setting name in "Name", and select "Continue"
	          (Other items are not mandatory)
	     (5) In "Device", USB, and select "Continue"
	         Example: With the EP-901A printer, "Model" is displayed as follows.
	             -----------------
	             "EP-901A USB #1(EP-901A)"
	             -----------------
	     (6) In "Model", set "EPSON-ESC/P-R Printer Driver for Linux", and
	         select "Add Printer"
	         Example: With the EP-901A printer, "Model" is displayed as follows.
	             -----------------
	             "EPSON EP-901A, EPSON-ESC/P-R Printer Driver for Linux (en,en,ja,nl,fr,de,it,pt,es,ko,zh_CN,zh_TW)"
	             -----------------
	     (6') Instead of steps (6), specify the PPD file directly from "Browse"
	          Example: With the EP-901A printer, you would specify the following file.
	              -----------------
	              /opt/epson-inkjet-printer-escpr/ppds/EPSON/EPSON-EP-901A-epson-escpr-en.ppd.gz
	              -----------------

This completes adding a printer for this software.


-------------------------------------------------------------------------------
5  Printing
-------------------------------------------------------------------------------
5.1  Items to be set
   The following items can be set with this software.

   * Ink
        Select color or monochrome printing.

   * Paper size
        Select the paper size for your document.

   * Print quality
        Select the media type and its print quality for your printing.


5.2  Details of setting items
   Followings is a list of setting options available for this software.

   * The meaning of the list items
     - Keyword          Text string used when specifying an option
     - Browser display  Text string displayed in the browser window for CUPS settings
     - Explanation      An explanation of the setting

   Ink
      Setting name for optional settings: Ink
      Setting name in the browser: Ink

      Keyword  | Browser display        | Explanation
      ---------+------------------------+------------------------------------
      COLOR    | Color                  | Color printing
      MONO     | Monochrome             | Monochrome printing

   Paper size
      Setting name for optional settings: PageSize or media
      Setting name in the browser: MediaSize

      Note: When adding multiple settings in the media option, put paper size
            at the beginning. For the media options, see the CUPS documentation.

   Print quality
      Setting name for optional settings: Quality
      Setting name in the browser: Quality

   Duplex printing
      Setting name for optional settings: Duplex
      Setting name in the browser: Duplex

   Paper source
      Setting name for optional settings: InputSlot
      Setting name in the browser: InputSlot

5.3  How to set

   There are three methods for print settings.

   * Using commands
        Use lpoptions. For details of the command, see lpoptions(1).

        Example: For the registered printer name ep901a
            To specify ink = COLOR, page size = A4, quality = PLAIN_NORMAL
        -----------------
        $ lpoptions -p ep901a -o ink=COLOR -o media=A4 -o quality=PLAIN_NORMAL
        -----------------

   * Use a browser
        Select "Configure Printer" from the printer window to display
        the setting window.

   * Specify options when printing
        As with lpoptions, set options with -o. For details of the commands,
        see lpr(1) and lp(1).

5.4  How to Print

   To print, as with the existing LPR, execute the lpr or lp commands.
   The formats that can be printed are as follows.

   * Postscript
   * PDF
   * Text
   * Various image files

   Example:
      When printing with the defaults
      -----------------
      $ lpr -P PRINTER_NAME <file>
      $ lp -d PRINTER_NAME <file>
      -----------------

      With printer_name = a950, ink = MONO, media = Legal
      -----------------
      $ lpr -P a950 -o ink=MONO -o media=Legal <file>
      $ lp -d a950 -o ink=MONO -o media=Legal <file>
      -----------------

The method of printing depends on the implementation of CUPS.
For more details, see the CUPS documentation.

-------------------------------------------------------------------------------
6  How to build in non-LSB distribution
-------------------------------------------------------------------------------
6.1  For Redhat based distributions
	6.1.1  Uncompress the src.rpm file
	   $ rpm2cpio epson-inkjet-printer-escpr-1.6.18-1lsb3.2.src.rpm | cpio -id
	   $ tar zxvf epson-inkjet-printer-escpr-1.6.18-1lsb3.2.tar.gz

	6.1.2  Configure and Create source tarball
	   $ cd epson-inkjet-printer-escpr-1.6.18
	   $ ./bootstrap && ./configure --prefix=/usr && make dist

	6.1.3  Create the directory for RPM packages
	   $ mkdir -p ~/rpmbuild/SOURCES
	   $ mkdir -p ~/rpmbuild/SPECS
	   $ mkdir -p ~/rpmbuild/BUILD
	   $ mkdir -p ~/rpmbuild/RPMS
	   $ mkdir -p ~/rpmbuild/SRPMS
	   
	   You can specifiy the another directory in ~/.rpmmacros
	   
	6.1.4  Copy the source tarball and the spec file to the directory
	   $ cp epson-inkjet-printer-escpr-1.6.18.tar.gz ~/rpmbuild/SOURCES/.
	   $ cp epson-inkjet-printer-escpr.spec ~/rpmbuild/SPECS

	6.1.5  Create RPM package
	   $ cd ~/rpmbuild/SPECS
	   $ rpmbuild -ba --clean epson-inkjet-printer-escpr.spec


6.2  For Debian based distributions
	6.2.1  Uncompress the src.rpm file
	   $ rpm2cpio epson-inkjet-printer-escpr-1.6.18-1lsb3.2.src.rpm | cpio -id
	   $ tar zxvf epson-inkjet-printer-escpr-1.6.18-1lsb3.2.tar.gz

	6.2.2  Configure
	   $ cd epson-inkjet-printer-escpr-1.6.18
	   $ ./bootstrap && ./configure --prefix=/usr

	6.2.3  Create DEB package
	   $ make
	   $ debuild -us -uc -b

6.3  For other distributions
	6.3.1  Uncompress the src.rpm file
	   $ rpm2cpio epson-inkjet-printer-escpr-1.6.18-1lsb3.2.src.rpm | cpio -id
	   $ tar zxvf epson-inkjet-printer-escpr-1.6.18-1lsb3.2.tar.gz

	6.3.2  Configure
	   $ cd epson-inkjet-printer-escpr-1.6.18
	   $ ./bootstrap && ./configure --prefix=/usr

	The location of CUPS filter and ppd files are decided automatically,
	but you can specify with the follwing configure options.
	
	* "--with-cupsfilterdir"    The location of CUPS filter
	                            ex) /usr/lib/cups/filter
	                            
	* "--with-cupsppddir"       The location of PPD files
	                            ex) /usr/share/ppd

	6.3.3  Make and Install
	   $ cd epson-inkjet-printer-escpr-1.6.18
	   $ make 
	   $ sudo make install


-------------------------------------------------------------------------------
7  Acknowledgments
-------------------------------------------------------------------------------
    We express our appreciation to the following people for cooperating with us
    in developing this product.

(In no particular order)
    Nobby N Hirano <nobby@nmail.hiug.ne.jp>
    Toshihiro Yamagishi <toshihiro@turbolinux.co.jp>
    Nozomi Sato <nozomi@esd.spr.epson.co.jp>

***End of file


NEWS
NEWS -- an overview of user-visible changes with each release

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.18                         (2017-12-19)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.17                         (2017-10-12)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.16                         (2017-09-13)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.15                         (2017-08-30)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.14                         (2017-07-27)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.13                         (2017-03-22)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.12                         (2017-02-13)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.11                         (2017-01-11)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.10                         (2016-11-16)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.9                          (2016-09-30)
----------------------------------------------------------------------
* Supported new models.

* Fixed a problem that Saturation value wasn't reflected 
  when using the lpr command.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.8                          (2016-09-14)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.7                          (2016-08-18)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.6                          (2016-06-27)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.5                          (2016-04-04)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.4                          (2016-02-17)
----------------------------------------------------------------------
* Supported new models.

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.3                          (2016-01-25)
----------------------------------------------------------------------
* Supported new models.


----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.2                          (2016-01-07)
----------------------------------------------------------------------
 * For following models, we changed the ppd description,
   "*DefaultMediaType" from PLAIN_DRAFT to PLAIN_NORMAL.

     - WF-3620 Series
     - WF-2740 Series
     - WF-4630 Series
     - WF-4640 Series
     - WF-5110 Series
     - WF-5190 Series
     - WF-5620 Series
     - WF-5690 Series
     - WF-6090 Series
     - WF-6530 Series
     - WF-6590 Series
     - WF-7110 Series
     - WF-7610 Series
     - WF-7620 Series
     - WF-8010 Series
     - WF-8090 Series
     - WF-8510 Series
     - WF-8590 Series
     - WF-R4640 Series
     - WF-R5190 Series
     - WF-R5690 Series
     - WF-R8590 Series
     - XP-211 214 216 Series
     - XP-212 213 Series
     - XP-215 217 Series
     - XP-310 Series
     - XP-312 313 315 Series
     - XP-410 Series
     - XP-412 413 415 Series

 * For following models, we added PLAIN_NORMAL and ENV_NORMAL as media type.
   
     - WF-M4011 Series
     - WF-M4015 Series
     - WF-M4095 Series
     - WF-M4521 Series
     - WF-M4525 Series
     - WF-M4595 Series

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.1                          (2015-10-23)
----------------------------------------------------------------------
 * Added ppd files of following models
     - PM-400 Series
     - L805 Series
     - WF-6530 Series

----------------------------------------------------------------------
epson-inkjet-printer-escpr-1.6.0                          (2015-09-29)
----------------------------------------------------------------------
 * Added ppd files of following models
     - EP-10VA Series
     - PX-M860F
     - PX-S860
     - WP-M4011 Series
     - WP-M4015 Series
     - WP-M4095 Series
     - WP-M4521 Series
     - WP-M4525 Series
     - WP-M4595 Series

 * Speeding up of multiple page printing


src/optBase.h
/*
 * Epson Inkjet Printer Driver (ESC/P-R) for Linux
 * Copyright (C) 2002-2005 AVASYS CORPORATION.
 * Copyright (C) Seiko Epson Corporation 2002-2013.
 *
 *  This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA.
 */

#ifdef HAVE_CONFIG_H
#  include <config.h>
#endif

#define END_ARRAY -1	

#ifndef N_
#define N_(magic_word)      (magic_word)
#endif

typedef struct _OPTION_DATA
{
	int value;
	char *rsc_name;
	char *x_name;
	char *x_name_jp;
} OPTION_DATA;


OPTION_DATA mediaSizeData[] = {
    {PM_MSID_A4,            "A4",       N_("A4  210x297 mm")},
    {PM_MSID_LETTER,        "Letter",   N_("Letter  8 1/2x11 in")},
    {PM_MSID_LEGAL,         "Legal",      N_("Legal  8 1/2x14 in")},
    {PM_MSID_A5,            "A5",       N_("A5  148x210 mm")},
    {PM_MSID_A6,            "A6",       N_("A6  105x148 mm")},
    {PM_MSID_B5,            "B5",       N_("B5  182x257 mm")},
    {PM_MSID_EXECUTIVE,     "EXE",      N_("Executive  7 1/4x10 1/2 in")},
    {PM_MSID_HALFLETTER,    "HLT",      N_("Half Letter  5 1/2x8 1/2 in")},
    {PM_MSID_PANORAMIC,     "PANORAM",  N_("Panoramic 210x594 mm")},
    {PM_MSID_TRIM_4X6,      "4X6",      N_("4x6 100x152 mm")},
    {PM_MSID_4X6,           "4X6FULL",  N_("Photo Paper 4x6 in  No Perforations")},
    {PM_MSID_5X8,           "INDEX5",   N_("Index card  5x8 in")                },
    {PM_MSID_8X10,          "8x10",   N_("Index card  8x10 in")               },
    {PM_MSID_10X15,         "PP100",    N_("Photo Paper 100x150 mm")            },
    {PM_MSID_200X300,       "PP200",    N_("Photo Paper 200x300 mm")            },
    {PM_MSID_L,             "L",        N_("3.5x5 in  89x127 mm")               },
    {PM_MSID_POSTCARD,      "Postcard", N_("Post Card  100x148 mm")             },
    {PM_MSID_DBLPOSTCARD,   "DoublePostcard",      N_("Double Post Card  200x148 mm")      },
    {PM_MSID_ENV_10_L,      "ENV10_L",  N_("Envelope #10(Landscape) 4 1/8x9 1/2 in")      },
    {PM_MSID_ENV_C6_L,      "ENVC6_L",  N_("Envelope C6(Landscape) 114x162 mm")           },
    {PM_MSID_ENV_DL_L,      "ENVDL_L",  N_("Envelope DL(Landscape) 110x220 mm")           },
    {PM_MSID_NEWEVN_L,      "ENV5X8_L", N_("Envelope(Landscape) 132x220 mm")              },
    {PM_MSID_CHOKEI_3,      "ENVC3",    N_("Envelope  120x235 mm")              },
    {PM_MSID_CHOKEI_4,      "ENVC4",    N_("Envelope  90x205 mm")               },
    {PM_MSID_YOKEI_1,       "ENVY1",    N_("Envelope  120x176 mm")              },
    {PM_MSID_YOKEI_2,       "ENVY2",    N_("Envelope  114x162 mm")              },
    {PM_MSID_YOKEI_3,       "ENVY3",    N_("Envelope  98x148 mm")               },
    {PM_MSID_YOKEI_4,       "ENVY4",    N_("Envelope  105x235 mm")              },
    {PM_MSID_2L,            "2L",       N_("5x7 in  127x178 mm")                },
    {PM_MSID_ENV_10,        "ENV10",    N_("Envelope #10  4 1/8x9 1/2 in")      },
    {PM_MSID_ENV_C6,        "ENVC6",    N_("Envelope C6  114x162 mm")           },
    {PM_MSID_ENV_DL,        "EnvDL",  	N_("Envelope DL  110x220 mm")           },
    {PM_MSID_NEWENV,        "ENV5X8",	N_("Envelope 132x220 mm")       	},
    {PM_MSID_BUZCARD_89X50, "NC",       N_("Business Card 89x50 mm")            },
    {PM_MSID_BUZCARD_89X55, "NC55",     N_("Business Card 89x50 mm")            },
    {PM_MSID_CARD_54X86,    "CARD",     N_("Card 54x86 mm")                       },
    {PM_MSID_BUZCARD_55X91, "NC91",     N_("Business Card 55x91 mm")              },
    {PM_MSID_ALBUM_A5,      "PAA5",     N_("Cover of Photo Album L 127 x 198 mm")     },
    {PM_MSID_ALBUM_L,       "PAL",      N_("Cover of Photo Album A5 210 x 321 mm") },
    {PM_MSID_PALBUM_L_L,    "PALL",     N_("P.Album L (Landscape) 127 x 89 mm") },
    {PM_MSID_PALBUM_2L,     "PA2L",     N_("P.Album 2L 127 x 178 mm")      },
    {PM_MSID_PALBUM_A5_L,   "PAA5L",    N_("P.Album A5 (Landscape) 210 x 148 mm") },
    {PM_MSID_PALBUM_A4,     "PAA4",     N_("P.Album A4 210 x 297 mm")           },
    {PM_MSID_4X7,           "4X7",      N_("16x9 wide size (102 x 181 mm)")},
    {PM_MSID_A3NOBI,        "A3+",      N_("A3+ 329x483 mm")          },
    {PM_MSID_A3,            "A3",       N_("A3  297x420 mm")                    },
    {PM_MSID_B4,            "B4",       N_("B4  257x364 mm")                    },
    {PM_MSID_USB,           "USB",      N_("US B(11x17 in)")},
    {PM_MSID_11X14,         "11X14",    N_("11x14 in")},
    {PM_MSID_B3,            "B3",       N_("B3 364 x 514 mm")},
    {PM_MSID_A2,            "A2",       N_("A2 420 x 594 mm")},
    {PM_MSID_USC,           "USC",      N_("US C 17 x 22 in")},
    {PM_MSID_10X12,	    "10X12",	N_("10x12 in / 254x305 mm")},
    {PM_MSID_12X12,	    "12X12",	N_("30 x 30 cm (12 x 12 in)")},
    {PM_MSID_USER,          "USER",     N_("User Defined")},
    {PM_MSID_UNKNOWN,       "UNKNOWN",  N_("Unknown Paper Size")},
    {END_ARRAY,                "",         ""                                      }
};


OPTION_DATA mediaTypeData[] = {

    { PM_MTID_PLAIN,         "PLAIN",    N_("plain papers")                      },
    { PM_MTID_360INKJET,     "FINE",     N_("360dpi Ink Jet Paper")              },
    { PM_MTID_IRON,          "IRON",     N_("Iron-On Cool Peel Transfer Paper")  },
//    { PM_MTID_PHOTOINKJET,   "SFINE",    N_("Photo Quality Ink Jet Paper")       },
    { PM_MTID_MATTE,   	     "SFINE",    N_("Photo Quality Ink Jet Paper")       }, /* Treated as "EPSON Matte" */
    { PM_MTID_PHOTOADSHEET,  "PADSHEET", N_("Photo Quality Self Adhesive Sheets")},
    { PM_MTID_MATTE,         "PMMATT",   N_("EPSON Matte")                       },
    { PM_MTID_PHOTO,         "GPPAPER",  N_("Photo Paper")            	 },
    { PM_MTID_PHOTOFILM,     "GFILM",    N_("Photo Quality Glossy Film")         },
    { PM_MTID_MINIPHOTO,     "MINIPHOTO",N_("Epson Photo Stickers")		 },
    { PM_MTID_OHP,           "OHP",      N_("Ink Jet Transparencies")            },
    { PM_MTID_BACKLIGHT,     "BLFILM",   N_("Ink Jet Back Light Film")           },
    { PM_MTID_PGPHOTO,       "PMPHOTO",  N_("EPSON Premium Glossy")        },
    { PM_MTID_PSPHOTO,       "PSGLOS",   N_("Epson Premium Semigloss")     },
    { PM_MTID_PLPHOTO,       "MCLP",     N_("Premium Luster Photo Paper")    	 },
    { PM_MTID_MCGLOSSY,      "PHOTOW",   N_("Glossy Paper - Photo Weight")         },
    { PM_MTID_ARCHMATTE,     "ARCMATT",  N_("Archival Matte Paper")              },
    { PM_MTID_WATERCOLOR,    "WATERCOLOR", N_("Watercolor Paper - Radiant White")},
    { PM_MTID_PROGLOSS,      "PROGLS", 	 N_("Professional Glossy Paper")},
    { PM_MTID_MATTEBOARD,    "MBOARD",   N_("Matte Board Paper")},
    { PM_MTID_PHOTOGLOSS,    "PHOTOGLS", N_("Photo Quality Glossy Paper")},
    { PM_MTID_SEMIPROOF,     "SEMIPROOF",N_("Dupont/EPSONSemigloss Proofing Paper-A")},
    { PM_MTID_SUPERFINE2,    "SFINE2",   N_("Photo Quality Ink Jet Paper Plus")},
    { PM_MTID_DSMATTE,       "DBSMATT",  N_("Double-Sided Matte Paper")},
    { PM_MTID_CLPHOTO,       "CLPP",     N_("ColorLife Photo Paper")},
    { PM_MTID_ECOPHOTO,      "EPHOTO",   N_("Economy Photo Paper")},
//    { PM_MTID_VELVETFINEART, "VELVET_FINE", N_("Velvet Fine Art Paper")},
    { PM_MTID_MATTE, 	     "VELVET_FINE", N_("Velvet Fine Art Paper")},	/* Treated as "EPSON Matte" */
    { PM_MTID_PROOFSEMI,     "PROOF_S",  N_("Proofing Paper Semimatte")},
    { PM_MTID_HAGAKIRECL,    "RCPC",     N_("Recycled Paper Postcard")},
    { PM_MTID_HAGAKIINKJET,  "IJPC",     N_("Ink Jet Postcard")},
    { PM_MTID_PHOTOINKJET2,  "PIJ2", 	 N_("Epson Photo Quality Ink Jet")},
    { PM_MTID_DURABRITE,     "EPP_P",    N_("DURABrite Photo Paper")},
    { PM_MTID_MATTEMEISHI,   "MMEISHI",  N_("Matte Business Card")},
    { PM_MTID_HAGAKIATENA,   "HAGAKIATENA",  N_("Post Card Address Side")},
    { PM_MTID_PHOTOALBUM,    "P_ALBUM",  N_("Photo Album")},
    { PM_MTID_PHOTOSTAND,    "P_BOARD",  N_("Photo Board Paper")},
    { PM_MTID_RCB,           "RCB",	 N_("RC-B")},
    { PM_MTID_PGPHOTOEG,     "PGPHOTOEG",N_("Premium Glossy Photo Paper_2")},
    { PM_MTID_ENVELOPE,      "ENV", 	 N_("Envelope")},
    { PM_MTID_PLATINA,       "PLATINA",  N_("Epson Ultra Glossy")},
    { PM_MTID_ULTRASMOOTH,   "USF_ART",  N_("UltraSmooth Fine Art Paper")},
    { PM_MTID_SFHAGAKI,      "SFHAGAKI",  N_("Super Fine Postcard")},
    { PM_MTID_PHOTOSTD,      "PHOTOSTD",  N_("Premium Glossy Photo Paper")},
    { PM_MTID_GLOSSYHAGAKI,  "GLOSSYHAGAKI", N_("Glossy Postcard")},
//    { PM_MTID_GLOSSYPHOTO,  "GLOSSYPHOTO",  N_("Glossy Photo Paper")},
    { PM_MTID_PGPHOTO,      "GLOSSYPHOTO",  N_("Glossy Photo Paper")},		/* Treated as "EPSON Premium Glossy" */
    { PM_MTID_GLOSSYCAST,	"GLOSSYCAST",  N_("Epson Photo")},
    { PM_MTID_BUSINESSCOAT, "BUSINESSCOAT", N_("Business Ink Jet Coat Paper")},
    { PM_MTID_CDDVD,         "CDDVD",    N_("CD/DVD")},
    { PM_MTID_CDDVDHIGH,     "CDDVDHIGH",     N_("CD/DVD Premium Surface")},
    { PM_MTID_PREPRINTED,    "PREPRINTED",     N_("Preprinted")},
    { PM_MTID_THICKPAPER,    "THICK",    N_("Thick Paper")},
    { PM_MTID_LETTERHEAD,    "LETTERHEAD",    N_("Letterhead")},
    { PM_MTID_RECYCLED,    "RECYCLED",    N_("Recycled")},
    { PM_MTID_COLORPAPER,    "COLORPAPER",    N_("Color")},
    { PM_MTID_CLEANING,      "CLEAN",    N_("Cleaning Sheet")},
    { PM_MTID_UNKNOWN,       "UNKNOWN",  N_("Unknown Paper Type")},
    { END_ARRAY,             "",         ""                       }
};

